src/components/common/Clickable.js

```diff
@@ -0,0 +1,44 @@
+import _ from 'lodash/fp'
+import { h } from 'react-hyperscript-helpers'
+import { containsUnlabelledIcon } from 'src/components/icons'
+import Interactive from 'src/components/Interactive'
+import TooltipTrigger from 'src/components/TooltipTrigger'
+import { forwardRefWithName, useLabelAssert } from 'src/libs/react-utils'
+
+
+const Clickable = forwardRefWithName('Clickable',
+  ({ href, as = (!!href ? 'a' : 'div'), disabled, tooltip, tooltipSide, tooltipDelay, useTooltipAsLabel, onClick, children, ...props }, ref) => {
```

Comment:
When we convert this component into TS, perhaps we can have stronger typing for HTML element strings like `'a'` and `'div'`, potentially with something like `type HTMLElement = 'a' | 'div'`, etc. (like [Scala case classes](https://docs.scala-lang.org/tour/case-classes.html))? It feels less error prone that way than having string literals scattered around.

Reply:
This is already done. The type signature requires that `as` be one of the keys of `JSX.IntrinsicElements`. [`JSX.InstrinsicElements` is defined by `@types/react`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e288812ba0306b2d818d4c89319418ff8c65cffc/types/react/index.d.ts#L3143-L3323) and includes all the HTML element names.

```ts
export type ClickableProps<T extends keyof JSX.IntrinsicElements = 'div'> = {
  as?: T
```

Reply:
I see. Thank you for the explanation. For someone who's not familiar with `keyof` (e.g. me as of 5 mins ago 😊), having that type alias`TagName` would've been helpful but I'm not suggesting doing that since Terra UI contributors will likely all need to be familiar with `keyof` eventually, in which case the additional verbosity is probably not worth it.

Reply:
I'm in favor of a type alias if it makes the error message clearer.

Reply:
Yea that could be a good reason for doing it.